### ISSUE TITLE
hostVerifier option

### ISSUE THREAD

**Issue by zhulduz** [mscdex/ssh2#363]
Hello,
I have a question abut hostVerifier option.

I guess this function should send hash and host_key, also I guess this function should wait until user set answer. Something like this

```coffeescript
hostVerifier: (hash, key, continue) ->
    dialog("RSA key fingerprint is #{hash}.
               Are you sure you want to continue connecting (yes/no)?")
    .done((answer) ->
       continue(answer)
     )
```

Or maybe I misunderstand something?
Thanks in advance.

**Comment by mscdex** [comment]
The `hostVerifier` is a synchronous function currently, so it needs to return true or false. I'd be open to a PR to change this though. To make it backwards compatible we could check if the function returned true or false. If it did, then we proceed as normal. If it returned undefined, we wait for the callback to be called. One tricky thing with this kind of change though is that we have to make sure the connection is still alive when the callback is called.

**Comment by zhulduz** [comment]
According, code structure, I don't see any way to wait for the callback when the function returned undefined. lots of code works synchronously (I guess it is strange, because it is complecated to interrupt some commands, like 'ls -R').

**Comment by mscdex** [comment]
Async host fingerprint verification support has now landed in master mscdex/ssh2@3beb9bdc817003ce1fb7bd93b5de409f7f14d524 and mscdex/ssh2-streams@7c5621641ee6d89f8c7ec69225bc65019269517e.
